### PR TITLE
Avoid initial Flash Of Unstyled Content

### PR DESCRIPTION
**Pull request by bradleyayers**
This change ensures that popper is rendered prior to the first frame being painted. It fixes a race condition where the first call to `requestAnimationFrame` is "too late" and is not scheduled before the browser paints the page.

### CODE REVIEW

File: src/utils/debounce.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    timeoutDuration = 1;
+    break;
+  }
+}
+
+function microtaskDebounce(fn) {
+  let called = false;
+  let i = 0;
+  const elem = document.createElement('span');
+  const observer = new MutationObserver(() => {
```
Comment (FezVrasta) [inline src/utils/debounce.js]: I don't get what's the purpose of this observer...

**Comment by bradleyayers** [comment]
@FezVrasta thanks for the quick review, I'll investigate and fix those issues.

**Comment by bradleyayers** [comment]
@FezVrasta I'm having test failures from v1-dev:

```
[core] inits a popper with boundariesElement set to viewport, the popper is not in the viewport
```

**Comment by FezVrasta** [comment]
@bradleyayers yes I'm aware of that problem, there's a PR that should fix it but I need to finish it.

I'll hold on merging this PR until `v1-dev` is green again. Sorry!

**Comment by bradleyayers** [comment]
@FezVrasta any progress on this?

**Comment by FezVrasta** [comment]
No sorry I'm having some busy weeks. As soon i get some free time I'll try to finish the other PR.

If you have some free time you may want to give it a shot and try to fix the test that is not passing 😇 (#92)

**Comment by FezVrasta** [comment]
Manually merged, thanks.

**Comment by FezVrasta** [comment]
@bradleyayers the build fails on IE10, are you able to take a look?

https://travis-ci.org/FezVrasta/popper.js/builds/173897674

**Pull request closed without merging**